### PR TITLE
Sim G4 Geometry from dd4hep Tracked Flag

### PR DESCRIPTION
**Pull request by ianna**
#### PR description:

* Track g4GeometryDD4hepSource parameter
* Clean up DDDWorld API

#### if this PR is a backport please specify the original PR:

Before submitting your pull requests, make sure you followed this checklist:
- verify that the PR is really intended for the chosen branch
- verify that changes follow [CMS Naming, Coding, And Style Rules](http://cms-sw.github.io/cms_coding_rules.html)
- verify that the PR passes the basic test procedure suggested in the [CMSSW PR instructions](https://cms-sw.github.io/PRWorkflow.html)

### CODE REVIEW

**Comment by ianna** [comment]
@fabiocos - FYI

**Comment by ianna** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by fabiocos** [comment]
@ianna thank you, I think this can be helpful in the transition period to avoid possible misconfigurations difficult to track

**Comment by fabiocos** [comment]
+1

File: SimG4Core/Geometry/interface/DDDWorld.h [hunk 1]
```diff
@@ -29,6 +27,7 @@ class DDDWorld {
   G4VPhysicalVolume *GetWorldVolumeForWorker() const { return m_world; }
 
 private:
+  void SetAsWorld(G4VPhysicalVolume *pv);
```
Comment (fabiocos) [inline SimG4Core/Geometry/interface/DDDWorld.h]: @ianna @civanch the idea behind removing the ```static``` declaration is that in any case this is called only at initialization level, before the event chain?

Comment (ianna) [reply SimG4Core/Geometry/interface/DDDWorld.h]: yes. It's DDDWorld self-initialisation - the method became private.

**Pull request merged**